src/graphql/account.rs

```diff
@@ -418,6 +418,60 @@ impl AccountMutation {
         update_jwt_expires_in(expires_in)?;
         Ok(time)
     }
+
+    /// Retrieves the user's language selection by username.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn language(&self, ctx: &Context<'_>, username: String) -> Result<Option<String>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        let account = map
+            .get(&username)?
+            .ok_or_else::<async_graphql::Error, _>(|| "Account not found".into())?;
+
+        Ok(account.language)
+    }
+
+    /// Updates only the user's language setting, regardless of the user's role.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn update_language(
+        &self,
+        ctx: &Context<'_>,
+        username: String,
+        language: Option<UpdateLanguage>,
+    ) -> Result<String> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        let language = match language {
+            Some(update_language) => {
+                if let Some(new_language) = update_language.new {
+                    Some((update_language.old, Some(new_language)))
+                } else {
+                    return Err("New language must be provided and cannot be None.".into());
+                }
+            }
+            None => None,
+        };
+
+        map.update(
```

Comment:
The `update` method is intended for updating an account. We need to separate the exceptions for this `update_language` method and avoid having `update` handle them as well, because, from the user's perspective, the exceptions for `update_language` and `update` should be distinct.

Reply:
As I mentioned above, I have modified the code to throw an error if `language.new` is `None`.